### ISSUE TITLE
instance won't stop, even with -F

### ISSUE THREAD

**Issue by Rhinogradentia** [apptainer/singularity#4484]
### Version of Singularity:

3.4.0-1

### Expected behavior
Instance started with
`sudo singularity instance start image name`
can be stopped using this
`sudo singularity instance stop name`

### Actual behavior

Started an instance with:
```
sudo singularity instance start --writable image name
sudo singularity instance start --writable image name1
```
Tried to stop instances again in multiple ways:
```
sudo singularity instance stop name
sudo singularity instance stop --all
sudo singularity instance stop -s 15 name
sudo singularity instance stop -F name
sudo singularity instance stop -F --all
```
Every time I got:
`Stopping name instance of [path]/name (PID=xxxx)`

But afterward
`sudo singularity instance list`
still would list the instances as running and I could not start a new instance with the same name. 

EDIT: It seems even rebooting does not clear the running instances. But I don't think they are really running, but that there is somewhere only the information stored that they should be running. 

### Steps to reproduce behavior

Please see above. The definition file has debootstrap/trusty as OS and contains the installation of nginx and clones another github repository. Also it installs git, build-essential zlib1g-dev via apt, curls node.js from deb.noedsource.com (version 12.x) and installs it via apt. The nginx server is started in the startscript via service.
The image was build with the --sandbox parameter

#### OS distribution and version:

```
$ cat /etc/os-release 
PRETTY_NAME="Debian GNU/Linux 10 (buster)
NAME=Debian GNU/Linux
VERSION_ID=10

```

#### go version:

```
$ go version
go1.13 linux/amd64
```

#### How did you install Singularity

I followed the installation workflow from [https://sylabs.io/guides/3.4/user-guide/installation.html#install-on-linux](https://sylabs.io/guides/3.4/user-guide/installation.html#install-on-linux) and used the latest release from here.

**Comment by cclerget** [comment]
Hi @Rhinogradentia , I can't reproduce the issue you described. Please can you paste the definition file and also give more details about setup, like is your home directory is on a local filesystem, an encrypted or a network filesystem ?

**Comment by Rhinogradentia** [comment]
Hi @cclerget,
I will do this. Just for the moment - it is a local filesystem (laptop), not encrypted. 
In the meantime, I played around a little bit and a third "process" joined the first two which could not be "stopped" anymore.  This led me to the assumption that this was caused due to rebooting without stopping the instances beforehand.

**Comment by Rhinogradentia** [comment]
Here is the definition file:
```
BootStrap: debootstrap
OSVersion: trusty
MirrorURL: http://us.archive.ubuntu.com/ubuntu/

%runscript
    echo "This is what happens when you run the container..."

%post
    echo "Hello from inside the container"
    sed -i 's/$/ universe/' /etc/apt/sources.list
    apt-get update
    apt-get -y install vim nginx git build-essential zlib1g-dev curl samtools
    apt-get clean
    # install node.js - needed for jbrowse
    curl -sL https://deb.nodesource.com/setup_12.x | bash -
    apt-get install -y nodejs 
    # get jbrowse development version
    git clone https://github.com/gmod/jbrowse
    cd jbrowse
    git checkout 1.16.6-release
    # run setup
    ./setup.sh
    # mv jbrowse to nginx root directory
    cd .. && mv jbrowse /usr/share/nginx/html/
    # create data directory and copy data into it
    cd /usr/share/nginx/html/jbrowse && mkdir data
    
%startscript 
	service nginx start
```
I do everything of this in my home directory and I have sudo rights. Following is the list of unstoppable instances:
```
userX@host:~/Documents/singularity/jbrowse$ sudo singularity instance stop --all
Stopping jb instance of /home/userX/Documents/singularity/jbrowse/jbrowse (PID=7839)
Stopping jb1 instance of /home/userX/Documents/singularity/jbrowse/jbrowse (PID=10004)
Stopping jb2 instance of /home/userX/Documents/singularity/jbrowse/jbrowse.sif (PID=5432)

userX@host:~/Documents/singularity/jbrowse$ sudo singularity instance list
INSTANCE NAME    PID      IMAGE
jb               7839     /home/userX/Documents/singularity/jbrowse/jbrowse
jb1              10004    /home/userX/Documents/singularity/jbrowse/jbrowse
jb2              5432     /home/userX/Documents/singularity/jbrowse/jbrowse.sif
```
When I try to shell into the instance I get following:
```
userX@host:~/Documents/singularity/jbrowse$ sudo singularity shell instance://jb2
FATAL:   could not open proc directory /proc/5432: no such file or directory
userX@host:~/Documents/singularity/jbrowse$ sudo singularity shell instance://jb1
FATAL:   could not open proc directory /proc/10004: no such file or directory
userX@host:~/Documents/singularity/jbrowse$ sudo singularity shell instance://jb
FATAL:   could not open proc directory /proc/7839: no such file or directory

userX@host:~/Dokumente/singularity/jbrowse$ ls /proc
1     1617  1747   1843   2010   26    3428  38    4171  502  acpi         keys          sys
10    1618  1748   1845   2016   27    3429  3811  42    503  asound       key-users     sysrq-trigger
11    1620  1750   1846   20312  272   3431  3844  423   508  buddyinfo    kmsg          sysvipc
12    1625  1752   185    20314  288   3432  3858  424   509  bus          kpagecgroup   thread-self
1258  1628  1754   1854   20368  289   3434  3861  43    519  cgroups      kpagecount    timer_list
127   1633  1760   186    20380  2892  3435  39    44    521  cmdline      kpageflags    tty
129   1635  1761   1865   20395  2895  3437  3991  440   525  consoles     loadavg       uptime
130   1652  1762   1868   20401  29    3457  4     442   535  cpuinfo      locks         version
132   1659  1768   187    20520  291   3462  40    46    541  crypto       meminfo       vmallocinfo
133   1664  1779   188    20530  2988  35    4003  464   556  devices      misc          vmstat
134   1678  1780   1885   20532  3     3575  4021  468   561  diskstats    modules       zoneinfo
14    1689  1782   189    20551  30    36    4034  470   6    dma          mounts
1444  1693  1785   1894   20556  3079  3620  4046  471   606  driver       mtrr
1449  1699  1788   19     20558  3085  3678  4065  474   620  execdomains  net
1450  17    1791   190    21     31    3698  4077  477   66   fb           pagetypeinfo
1496  1702  1795   1923   22     32    37    4087  478   67   filesystems  partitions
15    1707  1797   19824  230    3261  370   4099  479   769  fs           sched_debug
1500  1708  18215  19843  231    33    3707  41    480   78   interrupts   schedstat
1502  1723  18216  19860  232    3389  371   4113  483   782  iomem        self
1510  1729  18217  19877  24     34    3728  4125  485   8    ioports      slabinfo
1512  1733  18218  19894  243    3413  3751  4141  488   9    irq          softirqs
1557  1737  18219  2      246    3419  3763  4143  497   939  kallsyms     stat
16    1741  1832   20     25     3422  3783  4155  498   950  kcore        swaps
```

**Comment by cclerget** [comment]
>  This led me to the assumption that this was caused due to rebooting without stopping the instances beforehand

@Rhinogradentia Indeed instance files was left in your home directory after reboot because the instance processes was killed by init process during reboot process without any possibilities for instances process to cleanup their files. It was fixed by #4372 but not included in 3.4, will be in next release 3.5.
As a workaround you can delete instances files manually `$HOME/.singularity/instances/sing` after reboot or use the version from master branch.

**Comment by Rhinogradentia** [comment]
Thank you very much!
